keras_nlp/tokenizers/byte_tokenizer.py

```diff
@@ -0,0 +1,197 @@
+        sequence is encountered (same behaviour as
+        https://www.tensorflow.org/api_docs/python/tf/strings/unicode_transcode).
+    replacement_char: int. Defaults to 65533. The replacement character to use
+        when an invalid byte sequence is encountered and when `errors` is set to
+        "replace" (same behaviour as
+        https://www.tensorflow.org/api_docs/python/tf/strings/unicode_transcode).
+
+    Examples:
+
+    Ragged outputs.
+    >>> inputs = ["hello"]
+    >>> tokenizer = keras_nlp.tokenizers.ByteTokenizer()
+    >>> tokenizer(inputs)
+    <tf.RaggedTensor [[104, 101, 108, 108, 111]]>
+
+    Dense outputs.
+    >>> inputs = ["hello"]
+    >>> tokenizer = keras_nlp.tokenizers.ByteTokenizer(sequence_length=8)
+    >>> tokenizer(inputs)
+    <tf.Tensor: shape=(1, 8), dtype=int32,
+                numpy=array([[104, 101, 108, 108, 111,   0,   0,   0]])>
+
+    Detokenization.
+    >>> inputs = ["hello"]
+    >>> tokenizer = keras_nlp.tokenizers.ByteTokenizer()
+    >>> tokenizer.detokenize(tokenizer.tokenize(inputs))
+    <tf.Tensor: shape=(1,), dtype=string, numpy=array([b'hello'], dtype=object)>
+
+    Detokenization with errors = "replace", replacement_char = 65533.
+    >>> inputs = ["he�llo"]
+    >>> tokenizer = keras_nlp.tokenizers.ByteTokenizer(
+        errors="replace", replacement_char=65533)
+    >>> tokenizer.detokenize(tokenizer.tokenize(inputs))
+    <tf.Tensor: shape=(1,), dtype=string, numpy=array([b'he\xef\xbf\xbdllo'],
+                dtype=object)>
+    """
+
+    def __init__(
+        self,
+        lowercase: bool = True,
+        sequence_length: int = None,
+        normalization_form: str = None,
+        errors: str = "replace",
+        replacement_char: int = 65533,
+        **kwargs,
+    ):
+        # Check dtype and provide a default.
+        if "dtype" not in kwargs or kwargs["dtype"] is None:
+            kwargs["dtype"] = tf.int32
+        else:
+            dtype = tf.dtypes.as_dtype(kwargs["dtype"])
+            if not dtype.is_integer:
+                raise ValueError(
+                    "Output dtype must be an integer type. "
+                    f"Received: dtype={dtype}"
+                )
+
+        # Check normalization_form.
+        if normalization_form not in (None, "NFC", "NFKC", "NFD", "NFKD"):
+            raise ValueError(
+                '`normalization_form` must be one of None, "NFC", "NFKC", '
+                '"NFD", "NFKD". Received: normalization_form='
+                f"{normalization_form}"
+            )
+
+        # Check errors.
+        if errors not in ("strict", "replace", "ignore"):
+            raise ValueError(
+                '`errors` must be one of "strict", "replace", "ignore" '
+                f"Received: errors={errors}"
+            )
+
+        super().__init__(**kwargs)
+
+        self._dtype = kwargs["dtype"]
+        self._vocab = [i.tobytes() for i in np.arange(256, dtype=np.uint8)]
+        self._lowercase = lowercase
```

Comment:
For properties like this that were passed in, keep these public. `self.lowercase`